improve reactant highlighting in rxn drawing

NadineSchneider opened the pull request:
- define default colors for highlighting of reactants, atom or bonds
in the drawing options
- make highlighting colors customizable through the API

greglandrum commented on Code/GraphMol/MolDraw2D/MolDraw2D.cpp:
```diff
@@ -528,13 +530,10 @@ void MolDraw2D::drawReaction(const ChemicalReaction &rxn,
-    highlightColors.push_back(DrawColour(.8, .8, .3));
-    highlightColors.push_back(DrawColour(.8, .6, .8));
-
+    std::vector<DrawColour> colors = drawOptions().highlightColourPalette;
```
what about making `colors` a `std::vector<DrawColour> *` so that you can skip all the copying?

greglandrum commented on Code/GraphMol/MolDraw2D/Wrap/rdMolDraw2D.cpp:
```diff
@@ -62,7 +62,21 @@ std::map<int, double> *pyDictToDoubleMap(python::object pyo) {
   return res;
 }
+
+void pyListToColourVec(python::object pyo, std::vector<DrawColour> &res) {
```
I guess that, just to be careful, you want to do `res.clear()` here first.
You could also just do a `resize()` to the length of the pyo and then use assignment instead of push_back.

greglandrum commented on Code/GraphMol/MolDraw2D/Wrap/rdMolDraw2D.cpp:
```diff
@@ -62,7 +62,21 @@ std::map<int, double> *pyDictToDoubleMap(python::object pyo) {
+  for (unsigned int i = 0;
+       i < python::extract<unsigned int>(tList.attr("__len__")()); ++i) {
+    python::tuple tpl = python::extract<python::tuple>(tList[i]);
+    float r = python::extract<float>(tpl[0]);
```
it's too easy to make the obvious mistake here and provide the colours as values between 0 and 255 instead of 0 and 1. you should do a range check on each component and raise a `ValueError` if it's less than 0 or greater than 1.

greglandrum reviewed (changes requested):
a few small changes

NadineSchneider committed:
improve reactant highlighting in rxn drawing

- define default colors for highlighting of reactants, atom or bonds
in the drawing options
- make highlighting colors customizable through the API

greglandrum commented:
Thanks!
